examples/src/bin/glupload.rs

```diff
@@ -0,0 +1,310 @@
+        gl.BufferData(
+            gl::ARRAY_BUFFER,
+            (VERTEX_DATA.len() * mem::size_of::<f32>()) as gl::types::GLsizeiptr,
+            VERTEX_DATA.as_ptr() as *const _,
+            gl::STATIC_DRAW,
+        );
+
+        if gl.BindVertexArray.is_loaded() {
+            let mut vao = mem::uninitialized();
+            gl.GenVertexArrays(1, &mut vao);
+            gl.BindVertexArray(vao);
+        }
+
+        let pos_attrib = gl.GetAttribLocation(program, b"position\0".as_ptr() as *const _);
+        let color_attrib = gl.GetAttribLocation(program, b"color\0".as_ptr() as *const _);
+        gl.VertexAttribPointer(
+            pos_attrib as gl::types::GLuint,
+            2,
+            gl::FLOAT,
+            0,
+            5 * mem::size_of::<f32>() as gl::types::GLsizei,
+            ptr::null(),
+        );
+        gl.VertexAttribPointer(
+            color_attrib as gl::types::GLuint,
+            3,
+            gl::FLOAT,
+            0,
+            5 * mem::size_of::<f32>() as gl::types::GLsizei,
+            (2 * mem::size_of::<f32>()) as *const () as *const _,
+        );
+        gl.EnableVertexAttribArray(pos_attrib as gl::types::GLuint);
+        gl.EnableVertexAttribArray(color_attrib as gl::types::GLuint);
+    }
+
+    Gl { gl: gl }
+}
+
+fn create_pipeline() -> Result<gst::Pipeline, Error> {
+    gst::init()?;
+
+    let pipeline = gst::Pipeline::new(None);
+    let src =
+        gst::ElementFactory::make("videotestsrc", None).ok_or(MissingElement("videotestsrc"))?;
+    let upload = gst::ElementFactory::make("glupload", None).ok_or(MissingElement("glupload"))?;
+    let colorconvert = gst::ElementFactory::make("glcolorconvert", None)
+        .ok_or(MissingElement("glcolorconvert"))?;
+    let sink = gst::ElementFactory::make("appsink", None).ok_or(MissingElement("appsink"))?;
+
+    pipeline.add_many(&[&src, &upload, &colorconvert, &sink])?;
+    gst::Element::link_many(&[&src, &upload, &colorconvert, &sink])?;
+
+    let appsink = sink
+        .dynamic_cast::<gst_app::AppSink>()
+        .expect("Sink element is expected to be an appsink!");
+
+    appsink.set_property("enable-last-sample", &false.to_value())?;
+    appsink.set_property("emit-signals", &false.to_value())?;
+    appsink.set_property("max-buffers", &1u32.to_value())?;
+
+    let mediatype = format!(
+        "video/x-raw({}), format = (string) {}",
+        &gst_gl::CAPS_FEATURE_MEMORY_GL_MEMORY.to_string(),
+        &CAPS_FORMAT.to_string()
+    );
+    appsink.set_caps(&gst::Caps::from_string(&mediatype.as_str()));
+
+    appsink.set_callbacks(
+        gst_app::AppSinkCallbacks::new()
+            .new_sample(|appsink| {
+                let sample = match appsink.pull_sample() {
+                    None => return gst::FlowReturn::Eos,
+                    Some(sample) => sample,
+                };
+
+                let _buffer = if let Some(buffer) = sample.get_buffer() {
+                    buffer
```

Comment:
The buffer should probably be passed somewhere so that glutin can make use of it?

Reply:
yep, mapped and paint the texture from it